rotkehlchen/api/rest.py

```diff
@@ -2747,6 +2747,21 @@ def decode_evm_transaction(
             write_cursor.execute(
                 'DELETE FROM evm_transactions WHERE tx_hash=? AND chain_id=?',
                 (tx_hash, evm_chain.serialize_for_db()))
+        try:
```

Comment:
this looks corecct and was missing. Even docstring says it should do it.

One important point here is that perhaps we need to have these abstracted in proper functions and reused where needed. Probably this is already used inside the code elsewhere either completely or in parts, while this code is only on this api call.

Reply:
I looked around on friday and seems that we only call those two functions in a test. When querying them normally we do step by step fetching from etherscan. I'll add a note and we can refactor after the release